src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -333,7 +333,7 @@ internal Task ExtractRelativeToDirectoryAsync(string destinationDirectoryPath, b
             string name = ArchivingUtils.SanitizeEntryFilePath(Name, preserveDriveRoot: true);
             string? fileDestinationPath = GetFullDestinationPath(
                                                 destinationDirectoryPath,
-                                                Path.IsPathFullyQualified(name) ? name : Path.Join(Path.GetDirectoryName(destinationDirectoryPath), name));
+                                                Path.IsPathFullyQualified(name) ? name : Path.Join(destinationDirectoryPath, name));
```

Comment:
The `GetDirectoryName` unintentionally got here by copy-paste, and went unnoticed because of the trailing path separator on `destinationDirectoryPath`.

Reply:
Was this just unnecessary or was it breaking something?  If the latter, can we add a test?

Reply:
It is unnecessary, and was introduced in https://github.com/dotnet/runtime/pull/85378 which was merged two days ago.